### PR TITLE
KAFKA-4222 / KAFKA-8700 / KAFKA-5566: Wait for state to transit to RUNNING upon start

### PR DESCRIPTION
**Pull request by guozhangwang**
I looked into the logs of the above tickets, and I think for a couple fo them it is due to the fact that the threads takes time to restore, or just stabilize the rebalance since there are multi-threads. Adding the hook to wait for state to transit to RUNNING upon starting. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@cpettitt-confluent @mjsax for reviews.

File: streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java [hunk 1]
```diff
@@ -261,6 +262,16 @@ private KafkaStreams createCountStream(final String inputTopic,
...
+
+            try {
+                TestUtils.waitForCondition(
+                    () -> stateListener.mapStates.containsKey(KafkaStreams.State.RUNNING),
+                    "Did not start successfully after " + TestUtils.DEFAULT_MAX_WAIT_MS + " ms"
+                );
+            } catch (final InterruptedException e) {
+                if (!stateListener.mapStates.containsKey(KafkaStreams.State.RUNNING))
+                    fail("Did not start successfully");
+            }
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java:275]: Why not use `startKafkaStreamsAndWaitForRunningState()` here?

File: streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java [hunk 2]
```diff
@@ -446,7 +457,8 @@ public void concurrentAccesses() throws Exception {
             windowStoreName,
             streamsConfiguration);
 
-        kafkaStreams.start();
+        StreamsTestUtils.startKafkaStreamsAndWaitForRunningState(kafkaStreams);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java]: Nit: add static import

**Review by mjsax — COMMENTED** [review]
Overall LGTM.

File: streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java (hunk 1 above)
Comment (cpettitt-confluent) [reply streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java:275]: +1

**Review by cpettitt-confluent — APPROVED** [review]
LGTM once @mjsax's comment on using `startKafkaStreamsAndWaitForRunningState` is resolved.

File: streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java:275]: The streams instance already has a registered state listener above for further testing purposes, so I have to reuse that one than calling `startKafkaStreamsAndWaitForRunningState` to register a new one.

Comment (cpettitt-confluent) [reply streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java:275]: Ahh! https://github.com/apache/kafka/pull/7500 allows you to wait for running state while still using an existing listener. But still waiting for it to get merged.

Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java:275]: I left a comment on #7500, basically I think we do not need two functions in IntegrationTestUtils and StreamsTestUtils.

**Comment by guozhangwang** [comment]
Cherry-picked to 2.4.

**Pull request merged**